### PR TITLE
Skip deleting options if columnar.options is already dropped

### PR DESCRIPTION
**Pull request by onurctirtir**
Drop Citus might cascade to `columnar.options` before dropping a
columnar table. In that case, we were getting below error when opening
`columnar.options` to delete records for the columnar table that we
are about to drop: "ERROR:  could not open relation with OID 0".

I somehow reproduced this bug more easily when testing pg upgrade
scenarios, probably because OID's of columnar metadata tables gets
re-written when upgrading pg. That is why I added the test to
after_pg_upgrade_schedule, which easily reproduces the bug after
reverting the fix done in `columnar_metadata.c`

DESCRIPTION: Fixes a bug that could break `DROP SCHEMA/EXTENSON` commands when there is a columnar table

### CODE REVIEW

File: src/backend/columnar/columnar_metadata.c [hunk 1]
```diff
@@ -335,8 +335,13 @@ DeleteColumnarTableOptions(Oid regclass, bool missingOk)
...
 
-	Relation columnarOptions = relation_open(ColumnarOptionsRelationId(),
-											 RowExclusiveLock);
+	Relation columnarOptions = try_relation_open(ColumnarOptionsRelationId(),
+												 RowExclusiveLock);
+	if (columnarOptions == NULL)
+	{
+		/* extension has been dropped */
+		return false;
+	}
```
Comment (onurctirtir) [inline src/backend/columnar/columnar_metadata.c:344]: as we do in `DeleteMetadataRows` ...

File: src/test/regress/expected/upgrade_columnar_after.out [hunk 2]
```diff
@@ -247,3 +247,16 @@ BEGIN;
 	-- since we run "after schedule" twice, rollback the transaction
 	-- to avoid getting "table already exists" errors
 ROLLBACK;
+BEGIN;
+  -- Show that we drop columnar tables and columnar table access method
+  -- objects properly.
```
Comment (JelteF) [inline src/test/regress/expected/upgrade_columnar_after.out:252]:
```suggestion
  -- Show that we can still drop the extension after upgrading
```

The below output doesn't actually show that we are dropping columnar tables.

File: src/test/regress/after_pg_upgrade_schedule [hunk 3]
```diff
@@ -1 +1,5 @@
-test: upgrade_basic_after upgrade_columnar_after upgrade_type_after upgrade_ref2ref_after upgrade_distributed_function_after upgrade_rebalance_strategy_after upgrade_list_citus_objects upgrade_autoconverted_after
+test: upgrade_basic_after upgrade_type_after upgrade_ref2ref_after upgrade_distributed_function_after upgrade_rebalance_strategy_after upgrade_list_citus_objects upgrade_autoconverted_after
+
+# This attempts dropping citus extension, so please do not run in parallel
+# with any other tests.
```
Comment (onurctirtir) [inline src/test/regress/after_pg_upgrade_schedule:4]:
```suggestion
# This attempts dropping citus extension (and rollbacks), so please do
# not run in parallel with any other tests.
```

**Review by marcocitus — APPROVED** [review]

**Comment by onurctirtir** [comment]
(backported to those three branches, so removing labels)

**Pull request merged**